tvm/src/codegen/hlsc/codegen_vhls.h

```diff
@@ -15,7 +15,7 @@
 namespace TVM {
 namespace codegen {
 
-class CodeGenVivadoHLS final : public CodeGenHLSC {
+class CodeGenVivadoHLS : public CodeGenHLSC {
```

Comment:
See the comments above. Would be better to use the following inherence relation.
```
Stratus     Vivado
   \          /
        HLSC
```

Reply:
Stratus HLS backend shares a lot of common `VisitExpr_` and `VisitStmt_` function implementations with Vivado HLS backend, `CodeGenVivadoHLS` intentionally inherits `CodeGenVivadoHLS` to reduce code duplication.